ff4s/src/main/scala/ff4s/Store.scala

```diff
@@ -34,53 +35,37 @@ trait Store[F[_], State, Action] {
 
 object Store {
 
-  def apply[F[_]: Concurrent, State, Action](
-      initialState: State
-  )(
-      makeDispatcher: SignallingRef[F, State] => Action => F[Unit]
+  def apply[F[_]: Concurrent, State, Action](init: State)(
+      update: Action => State => (State, F[Option[Action]])
   ): Resource[F, Store[F, State, Action]] = for {
+    actionQ <- Queue.unbounded[F, Action].toResource
 
-    state0 <- SignallingRef.of[F, State](initialState).toResource
+    effectQ <- Queue.unbounded[F, F[Option[Action]]].toResource
 
-    dispatcher = makeDispatcher(state0)
+    stateSR <- SignallingRef.of[F, State](init).toResource
 
-  } yield (new Store[F, State, Action] {
-
-    override def dispatch(action: Action): F[Unit] = dispatcher(action)
-
-    override def state: Signal[F, State] = state0
-
-  })
-
-  def withRouter[F[_], State, Action](initialState: State)(
-      onUriChange: Uri => Action
-  )(
-      makeDispatcher: (
-          SignallingRef[F, State],
-          Router[F]
-      ) => Action => F[Unit]
-  )(implicit F: Async[F]) = for {
-
-    state0 <- SignallingRef.of[F, State](initialState).toResource
-
-    window = fs2.dom.Window[F]
-
-    router <- Router[F](window)
-
-    dispatcher = makeDispatcher(state0, router)
+    _ <- Stream
+      .fromQueueUnterminated(actionQ)
+      .evalMap(action => stateSR.modify(update(action)).flatMap(effectQ.offer))
```

Comment:
actually I think instead of an `effectQ` you could create an internal `Supervisor` and directly run the effects on that. It should be more-or-less equivalent to `parEvalMapUnorderedUnbounded`.